### ISSUE TITLE
[DBB V2] DB2 SQL Error: SQLCODE=-913, SQLSTATE=57033, SQLERRMC=00C9008E;00000210;DBBZ001 .DBBRSEQR.00000001, DRIVER=4.31.25

### ISSUE THREAD

**Issue by FALLAI-Denis** [IBM/dbb-zappbuild#405]
Hi,

Since the switch to DBB V2 and the management of the metadastore in Db2 z/OS, we are confronted with abend SQLCODE -913 / SQLERRMC 00C9008E on the` DBBRSEQR` tablespace which corresponds to the `DBB_SEQ_TABLE` table.

This happens when several builds are launched in parallel, and concerns builds launched after the 1st build.

The `DBB_SEQ_TABLE` table consists of a single row which appears to be a build counter.

We are at level of fix UI91246 /  PH53686.

**==> Could this problem be related to our adaptation of the build.groovy script?**

The relevant statement in our adapted build.groovy script is:

`def buildResult = metadataStore.createBuildResult(props.applicationBuildGroup, props.applicationBuildLabel)`

Should a particular action be taken after createBuildResult to free the resource linked to table `DBB_SEQ_TABLE`?

Detail of the problem:

```
11:55:41  com.ibm.dbb.build.BuildException: com.ibm.db2.jcc.am.SqlException: DB2 SQL Error: SQLCODE=-913, SQLSTATE=57033, SQLERRMC=00C9008E;00000210;DBBZ001 .DBBRSEQR.00000001, DRIVER=4.31.25
11:55:41       at com.ibm.dbb.metadata.jdbc.JDBCMetadataStore.getNextId(JDBCMetadataStore.java:791)
11:55:41       at com.ibm.dbb.metadata.jdbc.JDBCMetadataStore.getNextId(JDBCMetadataStore.java:751)
11:55:41       at com.ibm.dbb.metadata.jdbc.JDBCMetadataStore.createBuildResult(JDBCMetadataStore.java:838)
11:55:41       at com.ibm.dbb.metadata.jdbc.JDBCMetadataStore.createBuildResult(JDBCMetadataStore.java:819)
11:55:41       at java.lang.invoke.InterfaceHandle.invokeExact_thunkArchetype_L(InterfaceHandle.java:134)
11:55:41       at java.lang.invoke.CatchHandle.invokeExact_thunkArchetype_X(CatchHandle.java:76)
11:55:41       at java.lang.invoke.GuardWithTestHandle.invokeExact_thunkArchetype_X(GuardWithTestHandle.java:80)
11:55:41       at java.lang.invoke.GuardWithTestHandle.invokeExact_thunkArchetype_X(GuardWithTestHandle.java:80)
11:55:41       at java.lang.invoke.SpreadHandle.invokeExact_thunkArchetype_X(SpreadHandle.java:100)
11:55:41       at org.codehaus.groovy.vmplugin.v8.IndyInterface.fromCache(IndyInterface.java:318)
11:55:41       at java.lang.invoke.DirectHandle.invokeExact_thunkArchetype_L(DirectHandle.java:302)
11:55:41       at java.lang.invoke.AsTypeHandle.invokeExact_thunkArchetype_X(AsTypeHandle.java:49)
11:55:41       at java.lang.invoke.BruteArgumentMoverHandle.invokeExact_thunkArchetype_X(BruteArgumentMoverHandle.java:404)
11:55:41       at java.lang.invoke.CollectHandle.invokeExact_thunkArchetype_X(CollectHandle.java:130)
11:55:41       at java.lang.invoke.MutableCallSiteDynamicInvokerHandle.invokeExact_thunkArchetype_X(MutableCallSiteDynamicInvokerHandle.java:64)
11:55:41       at buildBPCE.initializeBuildProcess(buildBPCE.groovy:314)
```

And at the level of the Db2 master stc:

```
11.55.38 STC05917 ---- MONDAY,    28 AUG 2023 ----                                                     
11.55.38 STC05917  DSNT376I  -DB2$ PLAN=DISTSERV WITH  701                                             
   701                     CORRELATION-ID=db2jcc_appli                                                 
   701                     CONNECTION-ID=SERVER                                                        
   701                     LUW-ID=NE05AD0B.G62B.DDD08F73108D=289                                       
   701                                                                                                 
   701             THREAD-INFO=JENKINS:126.5.173.11:JENKINS:db2jcc_application:DYNAMIC:1               
   701             :*:<::126.5.173.11.1579.DDD08F73108D>                                               
   701                     IS TIMED OUT. ONE HOLDER OF THE RESOURCE IS PLAN=DISTSERV                   
   701             WITH                                                                                
   701                     CORRELATION-ID=db2jcc_appli                                                 
   701                     CONNECTION-ID=SERVER                                                        
   701                     LUW-ID=NE05AD0B.G629.DDD08EEC81E5=287                                       
   701                                                                                                 
   701             THREAD-INFO=JENKINS:126.5.173.11:JENKINS:db2jcc_application:DYNAMIC:1               
   701             :*:*                                                                                
   701                     ON MEMBER DB2$                                                              
11.55.38 STC05917  DSNT501I  -DB2$ DSNILMCL RESOURCE UNAVAILABLE  702                                  
   702                        CORRELATION-ID=db2jcc_appli                                              
   702                        CONNECTION-ID=SERVER                                                     
   702                        LUW-ID=NE05AD0B.G62B.DDD08F73108D=289                                    
   702                        REASON 00C9008E                                                          
   702                        TYPE 00000210                       
   702                        NAME DBBZ001 .DBBRSEQR.00000001     
```

**Comment by dennis-behm** [comment]
@FALLAI-Denis can you please funnel this through the official IBM support cases? Merci beaucoup.

**Comment by FALLAI-Denis** [comment]
Hi @dennis-behm 

A severity 2 incident was opened in May 2023 on the advice of Gil P., 

[[TS012878359](https://www.ibm.com/mysupport/5003p00002mY8yuAAC)](https://www.ibm.com/docs/en/db2/11.5?topic=management-lock-waits-timeouts)

Personally, I do not have access to this incident on the IBM site because I am not part of the system teams.

But we have not had a constructive response to this one.
I don't know who at IBM answered it.
Only people familiar with DBB's data model and what has been coded in Java classes could answer.

For me it's a DBB problem, but on the off chance I was wondering if this problem could be the consequence of a bad implementation of zAppbuild.

**Comment by FALLAI-Denis** [comment]
Hi,

What is the value of autocommit implemented in the DBB API?
Is it forced to false?

See:
* [Default JDBC autocommit modes](https://www.ibm.com/docs/en/db2-for-zos/12?topic=applications-default-jdbc-autocommit-modes)
* [Committing or rolling back JDBC transactions](https://www.ibm.com/docs/en/db2-for-zos/12?topic=applications-committing-rolling-back-jdbc-transactions)

**Comment by FALLAI-Denis** [comment]
Hi, 

FYI, a new CASE for this problem has been created: [TS014072726](https://www.ibm.com/mysupport/5003p00002oLud9AAC).
Previous was closed.

**Comment by M-DLB** [comment]
Thank you, @FALLAI-Denis 
As this problem is managed by the DBB Development team, I believe it would make sense to close this issue (issues opened here should only relate to the zAppBuild framework).
I hope you see no concern when closing this issue. 

Thank you!
